### PR TITLE
Create text component

### PR DESCRIPTION
**Pull request by allanhortle**

### CODE REVIEW

File: src/component/text/Text.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        children={children}
+    />;
+}
+
+Text.propTypes = {
+    /** {ClassName} */
+    className: PropTypes.string,
+
+    /** {ReactElement} */
+    element: PropTypes.oneOfType([
```
Comment (dxinteractive) [inline src/component/text/Text.jsx]: If this is the way to check for elements, best add it to decls/PropTypes.js, call it `ReactElement` and import it from there.

File: src/component/text/Text.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        PropTypes.element,
+        PropTypes.string,
+        PropTypes.func
+    ]),
+
+    /** {SpruceModifier} */
+    modifier: PropTypes.string,
+
+    /** Numeral format string. Will cause children to be passed through numeral.format */
```
Comment (dxinteractive) [inline src/component/text/Text.jsx]: Link to numeral format docs (and moment format docs) would be good here. Pretty sure all markdown is supported.

**Review by dxinteractive — CHANGES_REQUESTED** [review]

File: src/component/text/Text.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
+// @flow
+import React from 'react';
+import PropTypes from 'prop-types';
+import numeral from 'numeral';
+import moment from 'moment';
+import SpruceClassName from '../../util/SpruceClassName';
+
+export default function Text(props: Object): React.Element<any> {
```
Comment (dancoates) [inline src/component/text/Text.jsx]: Needs docs

**Review by dancoates — CHANGES_REQUESTED** [review]

File: src/decls/PropTypes.js [hunk 4]
```diff
@@ -6,8 +6,16 @@ const StampyPropTypes = {
...
 
+    element: PropTypes.oneOfType([
+        PropTypes.element,
+        PropTypes.string,
+        PropTypes.func
+    ]),
+
     htmlProps: PropTypes.object,
 
+    modifier: PropTypes.string,
```
Comment (dxinteractive) [inline src/decls/PropTypes.js]: Is this one necessary / being used? I think there's already a `StampyPropTypes.spruceModifier`, although the naming's a little inconsistent with the `modifier` prop name.

**Review by dancoates — APPROVED** [review]

**Review by dxinteractive — APPROVED** [review]

**Pull request merged**